src/main/webapp/resources/js/pages/announcement/ViewUnreadAnnouncement.jsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { Modal, Typography } from "antd";
+import { VisibilityProvider } from "../../contexts/visibility-context";
+import { formatDate } from "../../utilities/date-utilities";
+import Markdown from "react-markdown";
+import { LinkButton } from "../../components/Buttons/LinkButton";
+import { PriorityFlag } from "./PriorityFlag";
+/**
+ * Component to add a button which will open a modal to view a unread announcement.
+ * @param {function} ViewAnnouncement - the function that displays a unread announcement.
+ * @returns {*}
+ * @constructor
+ */
+const { Text } = Typography;
+const { confirm } = Modal;
+function ViewUnreadAnnouncementModal({ announcement, markAnnouncementAsRead }) {
+  const displayAnnouncement = () =>
+    confirm({
+      type: "info",
+      width: `60%`,
+      icon: <PriorityFlag hasPriority={announcement.priority} />,
+      okText: `Read`,
+      onOk() {
+        markAnnouncementAsRead(announcement.identifier);
+      },
+      title: (
+        <>
+          <Text strong>{announcement.title}</Text>
+          <br />
+          <Text type="secondary" style={{ fontSize: `.8em` }}>
+            {i18n(
+              "ViewUnreadAnnouncement.create.details",
+              announcement.user.username,
+              formatDate({ date: announcement.createdDate })
+            )}
+          </Text>
+        </>
+      ),
+      content: <Markdown source={announcement.message} />,
```

Comment:
I have a concern if the content is very long.  Perhaps wrap this in a div that has overflow scroll set on it for vertical with a max-height of window - 200px or something.

Reply:
Please see 2a7aff71e32004d63d7bf2d680ce7448ba74b604. I left the default at 300px, but will change to 200px if required.

Reply:
Please see c4dade373f91e51b7f98139f151c469f32229d61.